src/components/nav-icon-item/nav-icon-item.tsx

```diff
@@ -0,0 +1,63 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, FC, MouseEventHandler, ReactNode } from 'react'
+import { ElAnchorNavIconItem, ElButtonNavIconItem } from './styles'
+
+interface CommonNavIconItemProps {
+  /**
+   * The JSX element to render as the icon
+   */
+  icon: ReactNode
+
+  /**
+   * Whether the nav item is active
+   * @default false
+   **/
+  isActive?: boolean
+}
+
+interface NavIconItemAsButtonElementProps extends CommonNavIconItemProps, ButtonHTMLAttributes<HTMLButtonElement> {
+  href?: never
+  target?: never
+  ref?: never
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+
+interface NavIconItemAsAnchorElementProps extends CommonNavIconItemProps, AnchorHTMLAttributes<HTMLAnchorElement> {
+  href?: string
+  target?: string
+  rel?: string
+  onClick?: MouseEventHandler<HTMLAnchorElement>
+}
+
+export type NavIconItemProps = NavIconItemAsButtonElementProps | NavIconItemAsAnchorElementProps
+
+const isButtonElement = (props: NavIconItemProps): props is NavIconItemAsButtonElementProps => {
+  return !props.href
+}
+
+export const NavIconItem: FC<NavIconItemProps> = (props) => {
+  if (!isButtonElement(props)) {
+    const { isActive, onClick, icon, ...rest } = props ?? {}
+
+    return (
+      <ElAnchorNavIconItem
+        {...rest}
+        role="button"
+        data-state={isActive ? 'active' : undefined}
+        onClick={(e) => {
+          e.preventDefault()
```

Comment:
**issue:** I missed this in my original review, but we can't bake in `e.preventDefault()` here as that will prevent the link from being followed. It's up to the consumer to decide whether that's what they want or not.

Reply:
>  It's up to the consumer to decide whether that's what they want or not

right